plugin/plugin.go

```diff
@@ -151,6 +152,8 @@ func New(opts protogen.Options, request *pluginpb.CodeGeneratorRequest) (*ORMBui
 
 	if strings.EqualFold(params["engine"], "postgres") {
 		builder.dbEngine = ENGINE_POSTGRES
+	} else if strings.EqualFold(params["engine"], "clickhouse") {
```

Comment:
Since now we're introducing a new engine, its high time we modularize it by refactoring it. May be we can define an interface & include the methods which are different for different engines.

Reply:
@chinmayb Should i start working on this new refactoring you talked about?

Reply:
yes please

Reply:
sure, i will open a new PR when i'm done with that